fragment.go

```diff
@@ -1762,9 +1764,13 @@ func (f *fragment) importValue(columnIDs, values []uint64, bitDepth uint, clear
 	// Process every value.
 	// If an error occurs then reopen the storage.
 	if err := func() (err error) {
-		for i := range columnIDs {
+		for i := len(columnIDs) - 1; i >= 0; i-- {
 			columnID, value := columnIDs[i], values[i]
 			if smallWrite {
+				if _, ok := colSet[columnID]; ok {
```

Comment:
should we make the corresponding change in the non-smallWrite case? it seems like it's just as applicable there.

just as a sanity check: we *can't* have timestamps here, right? so we're not losing timestamp data about previous writes.

Reply:
we don't actually need it in the large write case, because it goes value by value, setting the bits directly, rather than collecting all the set bits and all the clear bits for all the values, and then processing all set bits followed by all clear bits.

I should actually test this case, though. Will do that.

No timestamps in BSI—maybe one day, but I guess we'll cross that bridge...

Reply:
Even so, given how expensive writes can be, it might be worth dropping the writes that would be overwritten by later writes, if that's a common thing.

Reply:
I'm not inclined to trade a (potentially large) allocation for speeding up some workloads (which I think are fairly rare). This might warrant more research, but since we have grand plans for improving value imports overall, I'd rather get this bug fixed without adding anything new to the large write path.

Reply:
good point.